lib/Parse/ParseStmt.cpp

```diff
@@ -1808,10 +1808,8 @@ ParserResult<Stmt> Parser::parseStmtRepeat(LabeledStmtInfo labelInfo) {
   SourceLoc whileLoc;
 
   if (!consumeIf(tok::kw_while, whileLoc)) {
-    diagnose(whileLoc, diag::expected_while_after_repeat_body);
-    if (body.isNonNull())
-      return body;
-    return makeParserError();
+    diagnose(body.getPtrOrNull()->getEndLoc(), diag::expected_while_after_repeat_body);
```

Comment:
The previous code was careful to handle the "null" case for "body", while this code will crash if "body" is null.